targets/pca10059.json

```
@@ -2,5 +2,6 @@
 	"inherits": ["nrf52840"],
 	"build-tags": ["pca10059"],
 	"linkerscript": "targets/pca10059.ld",
-	"flash-command": "nrfutil pkg generate --hw-version 52 --sd-req 0x0 --application {hex} --application-version 1 /tmp/tinygo_$$.zip && nrfutil dfu usb-serial -pkg /tmp/tinygo_$$.zip -p {port} -b 115200 && rm -f /tmp/tinygo_$$.zip"
+	"binary-format": "nrf-dfu",
+	"flash-command": "nrfutil dfu usb-serial -pkg {nrf-dfu} -p {port} -b 115200"
```

By changing this part to `{zip}`, the changes to `main.go` can be minimized.

```suggestion
	"binary-format": "nrf-dfu",
	"flash-command": "nrfutil dfu usb-serial -pkg {zip} -p {port} -b 115200"
```

Please refer to the `binary-format` and `{bin}` parts of esp32.

https://github.com/tinygo-org/tinygo/blob/release/targets/esp32.json#L18-L19

---

I continue to fail to see how the reference to this example will help me. The {bin} token has an file extension of ".bin".
The nrf-dfu token has a different extension. This is what breaks the logic of the code, IMO. 

Could you please sketch your proposal in more detail ?

---

Now I got it.

---

Partially reverted main and squashed.